public/main.js

```diff
@@ -25,63 +28,100 @@ deleteCompletedTodoBtn.onclick = function() {
     });
 };
 
+filterButtonAll.onclick = function() {
+    filterTodos("all");
+};
+filterButtonActive.onclick = function() {
+    filterTodos("active");
+};
+filterButtonComplete.onclick = function() {
+    filterTodos("complete");
+};
+
 function createTodo(title, callback) {
-    var createRequest = new XMLHttpRequest();
-    createRequest.open("POST", "/api/todo");
-    createRequest.setRequestHeader("Content-type", "application/json");
-    createRequest.send(JSON.stringify({
+    var dataToSend = JSON.stringify({
         title: title
-    }));
-    createRequest.onload = function() {
-        if (this.status === 201) {
-            callback();
-        } else {
-            error.textContent = "Failed to create item. Server returned " + this.status + " - " + this.responseText;
-        }
-    };
+    });
+    fetch("/api/todo/", {
```

Comment:
It looks like your functions using fetch have a lot in common. See whether you can refactor them to use common code?